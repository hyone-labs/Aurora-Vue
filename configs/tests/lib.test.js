import { render, screen, fireEvent } from '@testing-library/vue';
import * as Library from '@root/dist/lib/core-ui.umd';

describe('Lib Test', () => {
  it('builded with correct objects', () => {
    expect(Library).toHaveProperty('CoreUI');
    expect(Library).toHaveProperty('CoreUIComponents');
    // const { html } = render(Library);
    // expect(html()).toMatchSnapshot();
  });

  describe('when builded', () => {
    it('attach plugin correctly', async () => {
      const warn = jest.spyOn(console, 'warn');
      const error = jest.spyOn(console, 'error');
      const root = document.createElement('div');
      root.setAttribute('data-app', 'true');
      const handleClick = jest.fn();
      const exampleOfComponent = {
        template: '<h-button data-testid="example" @click="onClick">Event</h-button>',
        methods: {
          onClick: handleClick,
        },
      };

      const vueInstanceConfig = {
        global: {
          plugins: [Library.CoreUI],
        },
        container: document.body.appendChild(root),
      };

      const { html } = await render(
        exampleOfComponent,
        vueInstanceConfig,
      );

      expect(warn).not.toBeCalled();
      expect(error).not.toBeCalled();
      expect(html()).toMatchSnapshot();
      await fireEvent.click(screen.getByTestId('example'));
      expect(handleClick).toBeCalled();
      // expect(window.getComputedStyle(screen.getByTestId('example'))).toHaveProperty('background-color: blue');
    });
  });
});
