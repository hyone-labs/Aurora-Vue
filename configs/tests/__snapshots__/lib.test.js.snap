// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lib Test when builded attach plugin correctly 1`] = `
"<button class=\\"h-button h-button--color-primary h-button--size-small h-button--variant-contained\\" data-testid=\\"example\\">
  <!---->
  <!----><span class=\\"h-button__text\\">Deu Certo</span>
  <!---->
</button>"
`;
